src/Microsoft.OData.Core/UriParser/Parsers/ODataPathParser.cs

```diff
@@ -1062,6 +1130,16 @@ private bool TryCreateSegmentForOperation(ODataPathSegment previousSegment, stri
 
             this.TryBindKeySegmentIfNoResolvedParametersAndParenthesisValueExists(parenthesisExpression, returnType, resolvedParameters, segment);
 
+            if (tryBindingEscapeFunction && !this.TryBindEscapeFunction())

```

Comment:
>!this.TryBindEscapeFunction() [](start = 44, length = 29)

shouldn't this return false if bindescapefunction returns false, so the segment can be evaluated as a key?  i.e., "/Customers(2)/NS.FindOrderComposable(orderName='foo'):/xyz/abc:" currently succeeds, ignoring everything after Customers(2), and returns a path with 2 segments (Customers entity set and key) if there is no escape function defined that takes an order.